concordium-contracts-common-derive/src/derive.rs

```diff
@@ -500,17 +869,31 @@ pub fn impl_deserial(ast: &syn::DeriveInput) -> syn::Result<TokenStream> {
                     .zip(variant.fields.iter())
                     .map(|(name, field)| impl_deserial_field(field, name, &source))
                     .collect::<syn::Result<proc_macro2::TokenStream>>()?;
-                let idx_lit = syn::LitInt::new(i.to_string().as_str(), Span::call_site());
+
+                // Get the literal for the tag either from a 'tag' attribute of the index of the
+                // variant.
+                let (tag_lit, tag_span) = variant_attributes.tag.map_or_else(
+                    || {
+                        (
+                            syn::LitInt::new(i.to_string().as_str(), variant.span()),
+                            variant.ident.span(),
+                        )
+                    },
+                    |tag| (tag.value, tag.span),
+                );
```

Comment:
Please document here that collisions are checked a bit later. This was confusing at first.

Reply:
I have documented this.
The main reason is to avoid cloning. But they are relatively small so it is probably fine to clone if this reads better.